KubeArmor/feeder/policyMatcher.go

```diff
@@ -1040,23 +1054,36 @@ func (fd *Feeder) UpdateMatchedPolicy(log tp.Log) tp.Log {
 				}
 			}
 
+			fd.DefaultPosturesLock.Lock()
```

Comment:
why is this logic here?
why do we need to update fd.DefaultPostures during log generation?
does it cause additional overhead?

Reply:
WatchDefaultPosture only updates the values in `fd.DefaultPosture` when there are annotation related updates. When there is an log related update but that namespace has no default posture set due to no annotations, we set the namespace annotation to the global default posture for this log and all future log events in this namespace.

Should not have any significant overhead because if the we need to read for values anyway, if the value is already set we won't be assigning values here after that. The assignment happens once per namespace.